browser/src/Services/Explorer/ExplorerView.tsx

```diff
@@ -269,11 +266,77 @@ export interface IExplorerViewProps extends IExplorerViewContainerProps {
     updated: string[]
 }
 
-import { SidebarEmptyPaneView } from "./../../UI/components/SidebarEmptyPaneView"
+interface ISneakableNode extends IExplorerViewProps {
+    node: Node
+    selectedId: string
+}
 
-import { commandManager } from "./../CommandManager"
+const SneakableNode = ({ node, selectedId, ...props }: ISneakableNode) => {
+    const handleClick = () => {
+        props.onClick(node.id)
+    }
+
+    return (
+        <Sneakable callback={handleClick}>
+            <NodeView
+                node={node}
+                isSelected={node.id === selectedId}
+                isCreating={props.isCreating}
+                onCancelCreate={props.onCancelCreate}
+                onCompleteCreate={props.onCompleteCreate}
+                onCompleteRename={props.onCompleteRename}
+                isRenaming={props.isRenaming}
+                onCancelRename={props.onCancelRename}
+                updated={props.updated}
+                yanked={props.yanked}
+                moveFileOrFolder={props.moveFileOrFolder}
+                onClick={handleClick}
+            />
+        </Sneakable>
+    )
+}
+
+const ExplorerContainer = styled.div`
+    height: 100%;
+    ${enableMouse};
+`
 
 export class ExplorerView extends React.PureComponent<IExplorerViewProps, {}> {
+    private _virtualizedList: List
+
+    public componentDidUpdate(prevProps: IExplorerViewProps) {
+        if (
+            prevProps.isCreating !== this.props.isCreating ||
+            prevProps.isRenaming !== this.props.isRenaming
+        ) {
+            this._virtualizedList.recomputeRowHeights()
+            this._virtualizedList.forceUpdate()
+        }
+    }
+
+    public calculateHeight = (selectedId: string) => ({ index }: { index: number }) => {
+        const { nodes, isCreating, isRenaming } = this.props
+        const node = nodes[index]
+        const { id, name } = node
+        const isSelected = id === selectedId
+
+        const PADDING = 10
+        const fontSize = parseInt(this.props.fontSize, 10)
+        const rowSize = fontSize + PADDING
+
+        const renameInProgress = isRenaming.name === name && !isCreating && isSelected
+        const creationInProgress = isCreating && isSelected && !renameInProgress
+
+        switch (true) {
+            case renameInProgress:
+                return rowSize * 1.5
```

Comment:
I'm a bit concerned about these magic numbers here - `1.5` and `2.8`. If someone changes the styling of the node or the creation, they might not realize they need to update these here, and get difficult to find bugs. Along the lines of the above comment - is there a more deterministic way we can get these sizes?